xbmc/video/windows/GUIWindowVideoNav.cpp

```diff
@@ -153,19 +158,24 @@ bool CGUIWindowVideoNav::OnMessage(CGUIMessage& message)
             break;
           }
         }
-        if (i >= m_vecItems->Size() && url.GetOption("showinfo") == "true")
+        if (i >= m_vecItems->Size())
         {
-          // We are here if the item is filtered out in the nav window
-          std::string path = message.GetStringParam(0);
-          CFileItem item(path, URIUtils::HasSlashAtEnd(path));
-          if (item.IsVideoDb())
+          SelectFirstUnwatched();
```

Comment:
Shouldn't this just live outside of the above if/else? It's now in both branches that can happen.